settings/site/default.local.settings.php

```diff
@@ -45,8 +45,27 @@
  *
  * @see https://wiki.php.net/rfc/expectations
  */
-assert_options(ASSERT_ACTIVE, TRUE);
-Handle::register();
+/*
+ * If you are using php 8.3 and above assertions options
+ * usage is deprecated.
+ * @see https://www.drupal.org/node/3391611
+ *
+ * If you are using php 8.3 and less
+ * Drupal\Component\Assertion\Handle is deprecated.
+ * @see https://www.drupal.org/node/3105918
+ */
+if (phpversion() >= 8.3 && \Drupal::VERSION >= 10.2) {
```

Comment:
@deepakmishra2 I think we should simply use following code.
```suggestion
if (phpversion() >= 8.3 ) {
  ini_set('zend.assertions', 1);
} 
else {
  assert_options(ASSERT_ACTIVE, TRUE);
  assert_options(ASSERT_EXCEPTION, TRUE);
}
```

1. I think there's no need to check for Drupal version with PHP 8.3 because if anything deprecated in PHP 8.3, then we should use that, irrespective of Drupal version they use.
2. There's no need to use Handle class as it's doing the same thing i.e calling function `assert_options(ASSERT_EXCEPTION, TRUE);`. So, let's use as it is.
cc: @rajeshreeputra

Reply:
done with the changes , we were actually checking for 10.2 because those changes were specifically introduced in 10.2 version